Month1/Week2/Steam/Steam/Steam.cpp

```diff
@@ -0,0 +1,343 @@
+FGame Steam::CreateGame()
+{
+	system("CLS");
+
+	std::cout << "Please provide the details about the game to the best of your ability." << std::endl;
+
+	std::cout << "Game: ";
+	std::string GameName;
+	std::getline(std::cin, GameName);
+
+	std::cout << "Developer: ";
+	std::string GameStudioName;
+	std::getline(std::cin, GameStudioName);
+
+	std::cout << "Release Date (Year): ";
+	int GameYear;
+	std::cin >> GameYear;
+
+	GameYear = ValidateInt(GameYear, 1900, 2022);
+
+	std::cout << "Release Date (Month): ";
+	int GameMonth;
+	std::cin >> GameMonth;
+
+	GameMonth = ValidateInt(GameMonth, 1, 12);
+
+	std::cout << "Release Date (Day): ";
+	int GameDay;
+	std::cin >> GameDay;
+
+	GameDay = ValidateInt(GameDay, 1, 31);
+
+	FGame NewGame(GameName, GameStudioName, GameYear, GameMonth, GameDay);
+
+	return NewGame;
+}
+
+//-------------------------------------------------------------------------------------------
+//									MANAGING CATEGORIES
+//-------------------------------------------------------------------------------------------
+void Steam::ManageCategories()
+{
+	system("CLS");
+
+	std::cout << "Welcome to Category Management! What would you like to do?" << std::endl;
+	std::cout << "1 - Create a Category" << std::endl;
+	std::cout << "2 - Delete a Category" << std::endl;
+	int Option;
+	std::cin >> Option;
+
+	Option = ValidateInt(Option, 1, 2);
+
+	if (Option == 1)
+	{
+		CreateCategory();
+	}
+	else
+	{
+		DeleteCategory();
+	}
+
+	ActiveCommand = MenuCommand::MainMenu;
+}
+
+//-------------------------------------------------------------------------------------------
+void Steam::CreateCategory()
+{
+	system("CLS");
+	std::cin.clear();
+	std::cin.ignore(INT_MAX, '\n');
+
+	std::cout << "Please write the name of the category you would like to create: ";
+	std::string CategoryName;
+	if (std::cin.peek() == '\n') return;
+	std::cin >> CategoryName;
+
+	FCategory NewCategory(CategoryName);
+
+	std::cin.clear();
+	std::cin.ignore(INT_MAX, '\n');
+
+	if (Categories.AddCategory(NewCategory))
+	{
+		std::cout << "Success! You have created the category " << CategoryName << "!" << std::endl;
+		std::cin.ignore();
+	}
+	else
+	{
+		std::cout << "Unfortunately, there was an error creating your category. The category list is full." << std::endl;
+		std::cin.ignore();
+	}
+
+}
+
+//-------------------------------------------------------------------------------------------
+void Steam::DeleteCategory()
+{
+	system("CLS");
+
+	int CategoryCount = Categories.GetCategoryCount();
+	if (CategoryCount != 0)
+	{
+		ListCategories();
+		std::cout << "Which category would you like to delete?" << std::endl;
+		int CategoryChosen;
+		std::cin >> CategoryChosen;
+
+		CategoryChosen = ValidateInt(CategoryChosen, 0, CategoryCount);
+
+		if (Categories.DeleteCategory(CategoryChosen))
+		{
+			std::cout << "Success! That category has been deleted." << std::endl;
+			std::cin.ignore();
```

Comment:
The first ignore I added to each of these was added so that the screen didn't clear as it tried to return back to the Main Menu (since Main Menu inmediately clears as it is called). It _can_ be placed at the base scope of the function to avoid having to type it out three times, but it presents errors with subsequent inputs if it is not called twice in succession.